### PR TITLE
Deploy the new version of the website

### PR DESCRIPTION
**Pull request by bam241**
This should deploy the new version of the website.

The trigger is set on new tags, but it can be change.

We also need to add a token in CI to allow change on the `pyne.github.com` repo from CI

### CODE REVIEW

**Comment by gonuke** [comment]
If we are going to automate this, we should consider if/how to include the tag that triggers the build in the website itself so we know which version it is.

**Comment by gonuke** [comment]
Please rebase this for recent updates and let's consider how to move it forward.

**Comment by bam241** [comment]
@gonuke this "should" work, but it is hard to be sure...
this will trigger the website update on tag creation.
we could change it, to be triggered `master` branch change

**Comment by gonuke** [comment]
I'm not sure there is a way to test this without just trying it??

BTW, did you ever look at [doctr](https://github.com/drdoctr/doctr)?

**Comment by bam241** [comment]
I am thinking, we could add an option in the `Makefile` to push on a testing branch of the website repo, and see if it renders and works correctly, and when everything works set back the correct pushing mode.

never used `doctr`...

**Comment by bam241** [comment]
@gonuke we can see the results in `https://github.com/pyne/pyne.github.com/tree/ci_testing`
As it is set, every PR will trigger a website build that will be pushed on the `ci_testing` of the website repo.
I was thinking we could change this trigger on master merge. (allowing to check the website build)
and put a trigger on tags to push on the website `master` branch (to update the website)

File: .circleci/config.yml [hunk 1]
```diff
@@ -95,6 +95,30 @@ commands:
...
+          default: "push-test"
+    steps:
+      - pull_container:
+          build: "python2_dagmc_pymoab" 
+      - add_ssh_keys:
+          fingerprints:
+            - "04:66:ee:c2:be:fb:8d:c8:ec:5d:03:ce:ab:40:88:2c"
+      - run:
+          command: |
+              git config --global user.email "ci@pyne.io"
```
Comment (gonuke) [inline .circleci/config.yml]: Check whether these are necessary, and if so, we should decide what the best entries are.

File: .circleci/config.yml [hunk 2]
```diff
@@ -181,26 +223,42 @@ workflows:
...
   build_and_test:
     jobs:
-      - py3_build
-      - py3_test:
-          requires:
-            - py3_build
-    
-      - py2_build
-      - py2_test:
+        #      - py3_build
```
Comment (gonuke) [inline .circleci/config.yml]: uncomment this for final review

File: .circleci/config.yml [hunk 3]
```diff
@@ -95,6 +95,30 @@ commands:
           cd ~/repo/tests
           ./travis-run-tests.sh << parameters.flags >>
 
+  # Build and push PyNE website  
+  website_build_push:
+    description: "build PyNE website and push it"
+    parameters:
+      push_option:
+          type: string
+          default: "push-test"
```
Comment (gonuke) [inline .circleci/config.yml]:
```suggestion
          default: "test"
```

File: docs/Makefile [hunk 4]
```diff
@@ -123,6 +123,17 @@ push-release:
 	git commit -am "Pushed $(RELEASE) docs at $$(date)" && \
 	git push
 
+push-test:
+	cd $(BUILDDIR) && \
+	test -d $(DOCREPONAME) || git clone $(DOCREPOURL) $(DOCREPONAME) && \
+	cd $(DOCREPONAME) && \
+	git checkout -b ci_testing origin/ci_testing && \
```
Comment (gonuke) [inline docs/Makefile]:
```suggestion
	git checkout -b website_preview origin/website_preview && \
```

**Review by gonuke — COMMENTED** [review]
A few hopefully small changes.

File: .circleci/config.yml (hunk 1 above)
Comment (bam241) [reply .circleci/config.yml]: it is required... what are the best entries then ?

File: .circleci/config.yml (hunk 2 above)
Comment (bam241) [reply .circleci/config.yml]: done !

**Comment by bam241** [comment]
@gonuke when we decide what a good identity is, this should be good to be merged (or for an other review :) ) !
I implemented:
```
git config --global user.email "ci@pyne.io"
git config --global user.name "pyne-overseer"
```

**Comment by bam241** [comment]
the fingerprint key is not visible anymore :)

**Comment by bam241** [comment]
@gonuke I rebased this one, could you get a other look ?

File: .circleci/config.yml [hunk 5]
```diff
@@ -104,6 +104,30 @@ commands:
...
+          default: "test"
+    steps:
+      - pull_container:
+          build: "python2_dagmc_pymoab"
+      - add_ssh_keys:
+          fingerprints:
+            - $PYNE_KEY_FINGERPRINT
+      - run:
+          command: |
+              git config --global user.email "ci@pyne.io"
```
Comment (gonuke) [inline .circleci/config.yml]:
```suggestion
              git config --global user.email "pyne-dev@googlegroups.com"
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -104,6 +104,30 @@ commands:
...
+              git config --global user.name "pyne-overseer"
```
Comment (gonuke) [inline .circleci/config.yml]:
```suggestion
              git config --global user.name "report errors on github.com/PyNE/PyNE"
```

**Comment by gonuke** [comment]
I suggested some configuration info

**Comment by gonuke** [comment]
I think we need a `news` file and some documentation to let people know what the new automation does and under what circumstances

**Comment by gonuke** [comment]
ping @bam241 on news & documentation

**Comment by bam241** [comment]
@gonuke I added news file and documentation for the full circle.yml file.

I know it is not the object of the PR... but it was probably required...
I can split that into 2 PRs, if required... but I fear it will add some more delay....

File: .circleci/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
+PyNE-CI Documentation
+=====================
+
+
+PyNE Circle-CI use CIrcle_CI version 2.1 and is composed in 4 main sections:
+`executors`, `commands`, `jobs` and `workflow`.
+
+Executors
+---------
+The `executors` defines aliases to the environment to run the differents jobs, in our case
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
The `executors` define aliases to the environment to run the different jobs, in our case
```

File: .circleci/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+different docker images containing differents combinaisons of the optionnal
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
different docker images containing different combinations of the optional
```

File: .circleci/README.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+dependencies of PyNE (`MOAB`, `pyMOAB`, `DAGMC`) in one of the two supported
+PyTHON flavor (Python 2.7 or Python 3.6).
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
PyTHON flavors (Python 2.7 or Python 3.6).
```

File: .circleci/README.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Commands
+---------
+The `commands` section allows to define macro commands with arguments to be used
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
The `commands` section allows the definition of macro commands with arguments to be used
```

File: .circleci/README.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+in the differents jobs.
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
in the different jobs.
```

File: .circleci/README.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+`news_check`: check the presence of a new news file in the news folder
+
+
+`save_container`: allows to save the status of a container to be reloaded in an
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
`save_container`: saves the status of a container to be reloaded in an
```

File: .circleci/README.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+
+`pull_container`: pulls a previously saved (using the `save_container` command) status of container.
+
+    - `arguments`:
+        - `build` (string): build configuration parameters (has to match the build
+          argument used to save the container)
+
+
+`checkout_build`: checkout PyNE branch to be tested, build its and saves it
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
`checkout_build`: checks out PyNE branch to be tested, builds it and saves it
```

File: .circleci/README.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+usins the `save_container` command.
```
Comment (gonuke) [inline .circleci/README.md]:
```suggestion
using the `save_container` command.
```

**Review by gonuke — CHANGES_REQUESTED** [review]
Many grammar fixes in the README

File: .circleci/README.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
+PyNE-CI Documentation
+=====================
+
+
+PyNE Circle-CI use CIrcle_CI version 2.1 and is composed in 4 main sections:
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
PyNE Circle-CI use CIrcle_CI version 2.1 and is composed of 4 main sections:
```

File: .circleci/README.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+`executors`, `commands`, `jobs` and `workflow`.
+
+Executors
+---------
+The `executors` define aliases to the environment to run the different jobs, in our case
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
The `executors` define aliases to the environment to run the different jobs. In our case these are
```

File: .circleci/README.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+different docker images containing different combinations of the optional
+dependencies of PyNE (`MOAB`, `pyMOAB`, `DAGMC`) in one of the two supported
+PyTHON flavors (Python 2.7 or Python 3.6).
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
Python versions (Python 2.7 or Python 3.6).
```

File: .circleci/README.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Commands
+---------
+The `commands` section allows the definition of macro commands with arguments to be used
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
The `commands` section provides the definitions of macro commands with arguments to be used
```

Comment (kkiesling) [reply .circleci/README.md]: I think this wording is more of what you wanted here? Not sure "allows" is truly what is happening.

File: .circleci/README.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+in the different jobs.
+
+`news_check`: check the presence of a new news file in the news folder
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
`news_check`: checks the presence of a new news file in the news folder
```

File: .circleci/README.md [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+
+`checkout_build`: checks out PyNE branch to be tested, builds it and saves it
+using the `save_container` command.
+    
+    - `arguments`:
+        - `build` (string): used to identify a configuration
+        - `flags` (string): flags to be use when building PyNE
+
+
+`run_test`: pull a previous using `pull_container` command, then run the PyNE
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
`run_test`: pulls a previous container using `pull_container` command, then runs the PyNE
```

File: .circleci/README.md [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+nosetests using the provided `flag`.
+    
+    - `arguments`:
+        - `build` (string): used to identify a configuration
+        - `flags` (string): flags to be use when running the PyNE nosetests
+
+
+`website_build_push`: pull the Python 2.7 build with all the PyNE optionnal
+depedencies `python2_dagmc_pymaob` saved image (using `pull_container` command)
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
depedencies `python2_dagmc_pymoab` saved image (using `pull_container` command),
```

File: .circleci/README.md [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+
+`run_test`: pull a previous using `pull_container` command, then run the PyNE
+nosetests using the provided `flag`.
+    
+    - `arguments`:
+        - `build` (string): used to identify a configuration
+        - `flags` (string): flags to be use when running the PyNE nosetests
+
+
+`website_build_push`: pull the Python 2.7 build with all the PyNE optionnal
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
`website_build_push`: pull the Python 2.7 build with all the PyNE optional
```

File: .circleci/README.md [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+depedencies `python2_dagmc_pymaob` saved image (using `pull_container` command)
+build the website and push it to the `pyne.github.com` repo.
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
build the website, and push it to the `pyne.github.com` repo.
```

File: .circleci/README.md [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+    
+    - `arguments`:
+        - `push_option` (string: `test` or `root`): 
+            - `test` option will push the new built website to `website_preview`
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
            - `test` option will push the newly built website to `website_preview` 
```

File: .circleci/README.md [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+
+
+`website_build_push`: pull the Python 2.7 build with all the PyNE optionnal
+depedencies `python2_dagmc_pymaob` saved image (using `pull_container` command)
+build the website and push it to the `pyne.github.com` repo.
+    
+    - `arguments`:
+        - `push_option` (string: `test` or `root`): 
+            - `test` option will push the new built website to `website_preview` 
+            branch of the repo allowing previous allowing reviews. the push `root`
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
            branch of the repo allowing reviews. The push `root` 
```

File: .circleci/README.md [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+        - `build` (string): used to identify a configuration
+        - `flags` (string): flags to be use when running the PyNE nosetests
+
+
+`website_build_push`: pull the Python 2.7 build with all the PyNE optionnal
+depedencies `python2_dagmc_pymaob` saved image (using `pull_container` command)
+build the website and push it to the `pyne.github.com` repo.
+    
+    - `arguments`:
+        - `push_option` (string: `test` or `root`):
```
Comment (kkiesling) [inline .circleci/README.md:60]: I am very confused based on this specific part of the documentation what the options are? Are they `test`, `root`, and `master`??

File: .circleci/README.md [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+`website_build_push`: pull the Python 2.7 build with all the PyNE optionnal
+depedencies `python2_dagmc_pymaob` saved image (using `pull_container` command)
+build the website and push it to the `pyne.github.com` repo.
+    
+    - `arguments`:
+        - `push_option` (string: `test` or `root`): 
+            - `test` option will push the new built website to `website_preview` 
+            branch of the repo allowing previous allowing reviews. the push `root` 
+            version will push the website on the
+            - `master` branch of the repo, deploying a new version of the website
```
Comment (kkiesling) [inline .circleci/README.md]: confused by formatting here - why does this start at a new bullet?

File: .circleci/README.md [hunk 28]
```diff
@@ -0,0 +1,81 @@
...
+            version will push the website on the
+            - `master` branch of the repo, deploying a new version of the website 
+            in `pyne.io`
+
+
+Jobs
+----
+The `jobs` section defines all the different jobs used in the different
+workflows. For each built configuration two jobs have been defined, one to build
+PyNE, and one to run the `nosetests`. In addition to the the build and test
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
PyNE, and one to run the `nosetests`. In addition to the build and test
```

File: .circleci/README.md [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+jobs, two additional jobs have been created to build and push the website, one
+to the `website_test` branch of the `pyne.github.com` repo, and one to the
+`master` branch.
+
+
+Workflow
+--------
+The `workflow` section defines independant workflow triggers the differents
```
Comment (kkiesling) [inline .circleci/README.md]:
```suggestion
The `workflow` section defines independent workflow triggers the different
```

File: .circleci/config.yml [hunk 30]
```diff
@@ -104,6 +104,30 @@ commands:
           cd ~/repo/tests
           ./travis-run-tests.sh << parameters.flags >>
 
+  # Build and push PyNE website
+  website_build_push:
+    description: "build PyNE website and push it"
```
Comment (kkiesling) [inline .circleci/config.yml]: should this specify that the default "push" is the "test" push?

**Review by kkiesling — CHANGES_REQUESTED** [review]
Reviewed for grammar/english as requested. Many suggestions.

File: .circleci/README.md (hunk 26 above)
Comment (bam241) [reply .circleci/README.md:60]: this part is messed up... 
sorry about that....

File: .circleci/config.yml (hunk 30 above)
Comment (bam241) [reply .circleci/config.yml]: I have a slight preference to keep this description small, focused of the general behavior, but don't have a strong opinion about this... (as it is explain in the README file and the default value if specified 4 line below...)

But if anyone feel strongly about this, I can definitely change it!

Comment (kkiesling) [reply .circleci/config.yml]: If I did `website_build_push` I might expect it to actually update the website and then be confused why it was not. So specifying that it's a test push could be helpful? I'm not tied one way or the other either, but could help eliminate confusion.

Comment (bam241) [reply .circleci/config.yml]: does it look better ?

Comment (kkiesling) [reply .circleci/config.yml]: yes I think so

**Review by kkiesling — APPROVED** [review]
looks good to me now

**Comment by bam241** [comment]
thx for the review @kkiesling

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @bam241  - let's give this a try...

**Pull request merged**